docs/USAGE-GUIDE.md

```diff
@@ -500,6 +500,224 @@ legacy SHA-1 algorithms in CertificateVerify messages if TLS1.2 has been negotia
 |   "20200207"   |      X       |      X     |   X    |
 |   "rfc9151"    |              |      X     |        |
 
+## Sending and Receiving
+
+### Basic IO setup
+
+By default, s2n-tls sends and receives data using a provided file descriptor
+(usually a socket) and the read/write system calls. The file descriptor can be set with
+`s2n_connection_set_fd()`, or separate read and write file descriptors can be
+set with `s2n_connection_set_read_fd()` and `s2n_connection_set_write_fd()`.
+The provided file descriptor should be active and connected.
+
+In general the application is free to configure the file descriptor as preferred,
+including socket options. s2n-tls itself sets a few socket options:
+* If available, TCP_QUICKACK is used during the TLS handshake
+* If available and enabled via `s2n_connection_use_corked_io()`, TCP_CORK is
+used during the TLS handshake. TCP_NOPUSH or TCP_NODELAY may be used if TCP_CORK
+is not available.
+
+**Important Note:**
+If the read end of the pipe is closed unexpectedly, writing to the pipe will raise
+a SIGPIPE signal. **s2n-tls does NOT handle SIGPIPE.** A SIGPIPE signal will cause
+the process to terminate unless it is handled or ignored by the application.
+See the [signal man page](https://linux.die.net/man/2/signal) for instructions on
+how to handle C signals, or simply ignore the SIGPIPE signal by calling
+`signal(SIGPIPE, SIG_IGN)` before calling any s2n-tls IO methods.
+
+### Blocking or Non-Blocking?
+
+s2n-tls supports both blocking and non-blocking I/O.
+* In blocking mode, each s2n-tls I/O function will not return until it has sent
+or received all data.
```

Comment:
Isn't the receive partial? Like if i pass a buffer of 100 bytes, it's possible to only fill a single byte. The way this is phrased it makes it sound like it'll block until the application buffer is full.

Reply:
Yeah, but that's more of a weird quirk of s2n_recv (which you can turn off with the right config setting) than a feature of blocking vs non-blocking. It might complicate the blocking explanation to go into that here. But I can remove the "all".

Reply:
Discussed offline. It might block until the buffer is full ;) added a note to s2n_config_set_recv_multi_record.

But otherwise, rephrased to avoid making a statement about the behavior of s2n_recv.